salt/state.py

```diff
@@ -3082,6 +3082,18 @@ def call_chunk(self, low, running, chunks):
                 running[tag] = self.call(low, chunks, running)
         if tag in running:
             self.event(running[tag], len(chunks), fire_event=low.get("fire_event"))
+
+            for i, item_data in enumerate(
+                running[tag].pop("sub_state_run", {}).items()
+            ):
+                sub_tag, sub_return = item_data
+                sub_run_num = sub_return.get("__run_num__", i)
+                sub_return["__run_num__"] = float(
+                    "{}.{}".format(running[tag]["__run_num__"], sub_run_num)
+                )
+                sub_return["__sls__"] = running[tag]["__sls__"]
+                running[sub_tag] = sub_return
+
```

Comment:
Are we intentionally assuming at this point that the data here is well-formed? For instance, `item_data` will always contain a tuple (or, whatever) of data?

Reply:
yes, item_data will always be a tuple because it is just a result of using `enumerate` while iterating over a dictionary's items.